src/__tests__/utils.spec.ts

```diff
@@ -1,107 +1,184 @@
-import { UIDisplayData } from '@aics/simularium-viewer';
-import { getNewSelectionMap, convertMapToSelectionStateInfo } from '../utils';
-
-const mockVisibilitySelectionMap = {
-  agent1: ['agent1'],
-  agent2: ['agent2'],
-  agent3: [],
-};
-
-const mockUIDisplayData: UIDisplayData = [
-  {
-    name: 'agent1',
-    displayStates: [],
-    color: '#000000',
-  },
-  {
-    name: 'agent2',
-    displayStates: [],
-    color: '#000000',
-  },
-  {
-    name: 'agent3',
-    displayStates: [],
-    color: '#000000',
-  },
-];
+import {
+  convertMapToSelectionStateInfo,
+  getSelectionAfterChildCheckboxClick,
+  updateUserChangesAfterCheckboxClick,
+  getChildren,
+  makeEmptyUserSelections,
+} from '../utils';
 
 describe('utils for converting selection data types and handling selection actions', () => {
-  describe('getNewSelectionMap', () => {
-    it('it has each agent from uidisplaydata in visibility map and nothing else', () => {
-      const payload = getNewSelectionMap('agent1', mockVisibilitySelectionMap);
-      mockUIDisplayData.forEach(({ name }) => {
-        expect(payload).toHaveProperty(name);
+  describe('updateUserChangesAfterCheckboxClick', () => {
+    it('removes an agent with no children from selection if it was previously included', () => {
+      const result = updateUserChangesAfterCheckboxClick(
+        'agent_with_no_children',
+        [],
+        {
```

Comment:
nit: it would be nice to define this block in all these tests as `const prevState= {...}` and then send into updateUserChangesAfterCheckboxClick(name, prevState) just for readability